code/modules/clothing/clothing.dm

```diff
@@ -409,12 +409,14 @@ BLIND     // can't see anything
 			adjusted = DIGITIGRADE_STYLE
 		H.update_inv_w_uniform()
 
-	if(hastie)
-		hastie.on_uniform_equip(src, user)
+		if(hastie && H.w_uniform == src)

```

Comment:
how would a situation where the tie isn't attached to the suit but is being equipped arise?